api/src/main/java/org/openmrs/ObsReferenceRange.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.hibernate.search.annotations.DocumentId;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.Table;
+
+/**
+ * ObsReferenceRange is typically a reference range of an Observation 
+ * The reference range is created at the point of recording {@link Obs}
+ *
+ * @since 2.7.0
+ */
+@Entity
+@Table(name = "obs_reference_range")
+public class ObsReferenceRange extends BaseReferenceRange implements java.io.Serializable {
```

Comment:
This should implement the `OpenmrsData` interface. I'm also slightly concerned that the navigation here is the wrong-way around, i.e., I should be able to go from the Obs -> its reference range, right?

Reply:
@dkayiwa suggested that we don't need to touch Obs object. Unless I didn't understand your question @ibacher

Reply:
Hmmm... on this one, I think it makes sense to modify the Obs. Each Obs can have at most one reference range.

Reply:
I agree!

Reply:
In regards to implementing `OpenmrsData` interface, @dkayiwa suggested that `ObsReferenceRange` shouldn't be changed/retired, whereas `OpenmrsData` extends `Changeable` and `Voidable`.

Reply:
@dicksonmulli we can still implement the interfaces that @ibacher suggested and just ignore some of those fields. So yet go ahead and do as he suggested. 😊

Reply:
I would've thought they should be `Voidable` (à la `Observation`) and implement the `Creatable` interface.

Reply:
@dkayiwa We can't implement this interface and ignore the fields. Otherwise we will run into errors. I suggest we don't implement it if this object is not auditable and voidable.

Reply:
If we model it in a way that an Obs has a reference range property, then it can be metadata just like the concept property of Obs.

Reply:
@ibacher we should also implement `Auditable` interface right?

Reply:
@ibacher  `OpenmrsData` extends `Auditable`. Can I just implement `Creatable` and `Voidable` ?

Reply:
Yes. IIRC, the reason we have all these interfaces is because Orders and Obs are not intended to be `Changeable`, but are `Createable` and `Voidable`.